pkgs/development/cuda-modules/fixups/cuda/cuda_nvcc.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  backendStdenv,
+  cudaOlder,
+  setupCudaHook,
+}:
+prevAttrs: {
+  # Merge "bin" and "dev" into "out" to avoid circular references
+  outputs = builtins.filter (
+    x:
+    !(builtins.elem x [
+      "dev"
+      "bin"
+    ])
+  ) prevAttrs.outputs or [ ];
+
+  # Patch the nvcc.profile.
+  # Syntax:
+  # - `=` for assignment,
+  # - `?=` for conditional assignment,
+  # - `+=` to "prepend",
+  # - `=+` to "append".
+
+  # Cf. https://web.archive.org/web/20230308044351/https://arcb.csc.ncsu.edu/~mueller/cluster/nvidia/2.0/nvcc_2.0.pdf
+
+  # We set all variables with the lowest priority (=+), but we do force
+  # nvcc to use the fixed backend toolchain. Cf. comments in
+  # backend-stdenv.nix
+
+  postPatch =
+    let
+      nvvmReplace = lib.optionalString (cudaOlder "12.5") ''
+        --replace-fail \
+          '$(TOP)/$(_NVVM_BRANCH_)' \
+          "''${!outputBin}/nvvm" \
```

Comment:
H'm, has this forward-slash always been here? Seems dangerous, unless we never put `nvvmReplace` last

Reply:
Made a little less dangerous-looking in the latest.